src/tests/nativeaot/SmokeTests/HardwareIntrinsics/Program.cs

```diff
@@ -12,6 +13,36 @@ static int Main()
     {
         s_success = true;
 
+#if !DEBUG
+        Console.WriteLine("****************************************************");
+        Console.WriteLine("* Size test                                        *");
+        long fileSize = new System.IO.FileInfo(Environment.ProcessPath).Length;
+        Console.WriteLine($"* Size of the executable is {fileSize / 1024,7:n0} kB             *");
+        Console.WriteLine("****************************************************");
+
+        const int Meg = 1024 * 1024;
+        const int HalfMeg = Meg / 2;
+        long lowerBound, upperBound;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            lowerBound = 2 * Meg; // 2 MB
+            upperBound = 4 * Meg; // 4 MB
```

Comment:
Linux window is 1.5-2.0 MB. Should we shrink this to 3.0-3.5 MB?

Reply:
The OSX test will likely cross the 3.0 threshold downward soon. No point dealing with that - this is approximated by the Linux size test. OSX size is not in our 8.0 release criteria.